contribution-guide.md

```diff
@@ -0,0 +1,154 @@
+
+### Set up Eclipse
+
+As all Ballerina repositories are developed as Maven or Gradle projects, follow the steps below to import any Ballerina project to Eclipse similar to any Maven or Gradle project.
+
+1. In Eclipse, click **File** and then click **Import..**.
+2. Under the **Maven** option, select **Existing Maven Projects**. 
+3. Browse the file system and select the Maven module to open it.
+
+## Obtain the source code 
+
+1. Execute the below command to clone the ballerina-lang Git repo.
+```bash 
+git clone --recursive https://github.com/ballerina-platform/ballerina-lang.git
+```
+>**Tip:** If you have already forked the repository to your GitHub account, then execute the below command replacing 'YOUR-GITHUB-USERNAME' with your Git username.
+```bash 
+git clone --recursive https://github.com/<YOUR-GITHUB-USERNAME>/ballerina-lang.git
+```
+2. Execute the below command to update the Git submodules.
+```bash 
+git submodule update --init
+```
+## Build the project
+
+1. Navigate to the <BALLERINA-LANG_REPO> root directory and execute the below Gradle command.
+```bash 
+./gradlew clean build
+```
+2. Extract the built Ballerina distributions in the below locations.
+-  **runtime only:** <BALLERINA-LANG_REPO>/distribution/zip/jballerina/build/distributions/jballerina-<version>-SNAPSHOT.zip. 
+-  **runtime and tools (e.g., Ballerina Composer):** <BALLERINA-LANG_REPO>/distribution/zip/jballerina-tools/build/distributions/jballerina-tools-<version>-SNAPSHOT.zip.
+>**Note:** If you face an IOException error stating "Too many open files", this is due to the default number of possible open files being set to a lower number on your operating system than required for Ballerina to be compiled. You may have to increase the number of open files/file descriptors (FD) on your operating system to 1000000 (or higher).
+
+## Contribute to Ballerina grammar
+
+Ballerina grammar has been implemented using ANTLR plugin version 4.5.3. To get a basic understanding on ANTLR grammar syntax and concepts before working with Ballerina grammar, go to [Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference](https://www.oreilly.com/library/view/the-definitive-antlr/9781941222621/).
+
+[Ballerina grammar](https://github.com/ballerina-platform/ballerina-lang/tree/master/compiler/ballerina-lang/src/main/resources/grammar) consists of two files:
+
+- **BallerinaLexer.g4:** contains the lexer rules for Ballerina grammar. Lexer is responsible for tokenizing an input Ballerina source code.
+- **BallerinaParser.g4:** contains the parser rules. Parser listens to the token stream generated by the lexer. High level grammar productions/abstractions are defined in the parser using those tokens.
+
+Once a change is done to any of the grammar files, the lexer and the parser need to be re-generated. To generate the lexer and the parser, navigate to the <BALLERINA_LANG_REPO>/compiler/ballerina-lang/src/main/resources/grammar/ directory, and execute the below command. 
```

Comment:
Any specific reason to remove [1]?

[1] https://github.com/ballerina-platform/ballerina-lang/blame/master/CONTRIBUTING.md#L163

Reply:
It was missed by mistake. Added the following:

**Info:** The above command will autogenerate some Java classes. The Ballerina AST builder is written on top of the auto-generated `BallerinaParserBaseListener.java` class. Thus, if any new rules are added to the `BallerinaParser.g4`, above command will generate new methods in the `BallerinaParserBaseListener.java` and you need to override those newly-added methods inside the `BLangParserListener.java` accordingly.